torchgeo/datasets/vhr10.py

```diff
@@ -0,0 +1,544 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""NWPU VHR-10 dataset."""
+
+import os
+from typing import Any, Callable, Dict, List, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib import patches
+from PIL import Image
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import check_integrity, download_and_extract_archive, download_url
+
+
+def convert_coco_poly_to_mask(
+    segmentations: List[int], height: int, width: int
+) -> Tensor:
+    """Convert coco polygons to mask tensor.
+
+    Args:
+        segmentations (List[int]): polygon coordinates
+        height (int): image height
+        width (int): image width
+
+    Returns:
+        Tensor: Mask tensor
+    """
+    from pycocotools import mask as coco_mask  # noqa: F401
+
+    masks = []
+    for polygons in segmentations:
+        rles = coco_mask.frPyObjects(polygons, height, width)
+        mask = coco_mask.decode(rles)
+        # if len(mask.shape) < 3:
+        #     mask = mask[..., None]
+        mask = torch.as_tensor(mask, dtype=torch.uint8)
+        mask = mask.any(dim=2)
+        masks.append(mask)
+    if masks:
+        masks_tensor = torch.stack(masks, dim=0)
+    else:
+        masks_tensor = torch.zeros((0, height, width), dtype=torch.uint8)
+    return masks_tensor
+
+
+class ConvertCocoAnnotations:
```

Comment:
This is all quite a bit more complicated than it used to be. What was the reason why this was needed?

Reply:
This is mainly following torchvision's implementation. I assume this is more complex because they wanted to handle some edge cases.

Reply:
If those edge cases don't exist in VHR-10 and we don't have any plans of using this elsewhere I would skip the complexity until it's actually needed

Reply:
So I went over the implementation and while it is verbose I don't think there is any unnecessary complexity. Here we're mainly parsing the COCO dataset and returning the labels as a dict of tensors as we do for all other datasets. 

There are roughly 10 extra lines for handling cases where segmentation masks are not provided, which was a personal addition. Since MS COCO is a common spec for handling both object detection and instance segmentation annotations; I added this so that we would be covered if we ever encountered a MS COCO style remote sensing object detection dataset (only boxes). If required, I can remove these 10 lines.